### PR TITLE
feat: add support for `InvokeConstructor2` and `InvokeStaticMethod2` to front-end

### LINKED ISSUE #7768: Add `InvokeConstructor2` and `InvokeStaticMethod2` to frontend
Use the order: InvokeConstructor2, InvokeMethod2, and InvokeStaticMethod2 everywhere.

In WeededAst it should look like:

```scala
case class InvokeConstructor2(clazzName: Name.Ident, exps: List[Expr], loc: SourceLocation) extends Expr

case class InvokeMethod2(exp: Expr, methodName: Name.Ident, exps: List[Expr], loc: SourceLocation) extends Expr

case class InvokeStaticMethod2(clazzName: Name.Ident, methodName: Name.Ident, exps: List[Expr], loc: SourceLocation) extends Expr
```

In ResolvedAst it should look like:

```scala
case class InvokeConstructor2(clazz: Class[_], exps: List[Expr], loc: SourceLocation) extends Expr

case class InvokeMethod2(exp: Expr, methodName: Name.Ident, exps: List[Expr], loc: SourceLocation) extends Expr

case class InvokeStaticMethod2(clazz: Class[_], methodName: Name.Ident, exps: List[Expr], loc: SourceLocation) extends Expr
```

The idea is that the programmer will write:

```
import java.lang.Foo
.Foo.method()
```

After import resolution we will know that Foo is `Class[java.lang.Foo]`.

**Lets us also take this opportunity to rename `name` on `InvokeMethod2` to `methodName` everywhere.**

### PR DESCRIPTION
**Pull request by chanattan**
Fixes #7768.

### CODE REVIEW

**Review by mlutze — APPROVED** [review]
Looks good. Do we already have parsing?

**Comment by magnus-madsen** [comment]
> Looks good. Do we already have parsing?

That's next and harder.

File: main/src/ca/uwaterloo/flix/language/ast/KindedAst.scala [hunk 1]
```diff
@@ -166,7 +166,11 @@ object KindedAst {
 
     case class Do(op: Ast.OpSymUse, exps: List[Expr], tvar: Type.Var, loc: SourceLocation) extends Expr
 
-    case class InvokeMethod2(exp: Expr, name: Name.Ident, exps: List[Expr], mvar: Type.Var, tvar: Type.Var, evar: Type.Var, loc: SourceLocation) extends Expr
+    case class InvokeConstructor2(clazz: Class[_], exps: List[Expr], loc: SourceLocation) extends Expr
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/KindedAst.scala:169]: You will need mvar here too

File: main/src/ca/uwaterloo/flix/language/ast/KindedAst.scala [hunk 2]
```diff
@@ -166,7 +166,11 @@ object KindedAst {
...
+
+    case class InvokeMethod2(exp: Expr, methodName: Name.Ident, exps: List[Expr], mvar: Type.Var, tvar: Type.Var, evar: Type.Var, loc: SourceLocation) extends Expr
+
+    case class InvokeStaticMethod2(clazz: Class[_], methodName: Name.Ident, exps: List[Expr], loc: SourceLocation) extends Expr
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/KindedAst.scala:173]: and mvar here.

But you can add it when you need it :)

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 3]
```diff
@@ -1420,6 +1420,12 @@ object Resolver {
               }
           }
 
+        case NamedAst.Expr.InvokeConstructor2(clazzName, exps, loc) =>
+          val esVal = traverse(exps)(visitExp(_, env0))
+          mapN(esVal) {
+            es => ResolvedAst.Expr.InvokeConstructor2(Class.forName(clazzName.name), es, loc)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:1426]: Here we will need some more complex code that uses the import environment.

Maybe @mlutze can sketch the solution.

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 4]
```diff
@@ -1428,6 +1434,12 @@ object Resolver {
               ResolvedAst.Expr.InvokeMethod2(e, name, es, loc)
           }
 
+        case NamedAst.Expr.InvokeStaticMethod2(clazzName, methodName, exps, loc) =>
+          val esVal = traverse(exps)(visitExp(_, env0))
+          mapN(esVal) {
+            es => ResolvedAst.Expr.InvokeStaticMethod2(Class.forName(clazzName.name), methodName, es, loc)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:1440]: Same comment here as above.

**Comment by magnus-madsen** [comment]
I added some comments, but I will merge this now because it is useful scaffolding.

**Comment by magnus-madsen** [comment]
Next up: syntax and type inference.

**Pull request merged**